### ISSUE TITLE
WARN: won't propagate trace tags (err: max trace tags header len (0) reached)

### ISSUE THREAD

**Issue by martin-sucha** [DataDog/dd-trace-go#1180]
`PropagatorConfig.MaxTagsHeaderLen` option was introduced in https://github.com/DataDog/dd-trace-go/pull/1082, but we get warnings `WARN: won't propagate trace tags (err: max trace tags header len (0) reached)` after upgrading to 1.36.0 when we don't set this field, but we set other fields in the PropagatorConfig.

The default value is 512 and is set only if the PropagatorConfig is not present.

Other fields in PropagatorConfig are set to default values when zero.

Shouldn't `MaxTagsHeaderLen==0` mean that the default value of 512 be used so that the option is backwards-compatible?

In other words, shouldn't

```go
	if cfg.MaxTagsHeaderLen == 0 {
		cfg.MaxTagsHeaderLen = defaultMaxTagsHeaderLen
	}
```

be added to NewPropagator?
